homeassistant/components/zha/core/device.py

```diff
@@ -116,6 +117,22 @@ def set_device_id(self, device_id):
         """Set the HA device registry device id."""
         self._ha_device_id = device_id
 
+    @property
+    def device(self) -> zha_typing.ZigpyDeviceType:
```

Comment:
was there a reason to expose this? I was trying to keep this encapsulated in this class. Is it used anywhere in here?

Reply:
yes. `zha.core.channels.Channels` needs access to underlying zigpy device for channel instantiation. 
Alternatively, we could just pass `hass` and the actual `zigpy_device` to `Channels` class, but I need to find alternatives to differentiate between a new join and a rejoin.